### DISCUSSION TITLE
Site Maintenance Routines + Random Event Manager.

### DISCUSSION THREAD

**Discussion by ghost** [Foundation-19/Foundation-19#779] [Design]
# Site Maintenance Routines + Random Event Manager.
REM(Random Event Manager) is a system that generates site actions/events based on time in the round independently based on chance to proc things such as lights breaking, power outtages which in turn cause containment breaches.

## Abstract
As it is the Random Event Manager is just a system that causes a very limited array of things to happen from time to time.

Instead of it being an independent being that is active from the get-go of the round that goes haywire after a time with no retort possible from the players. I would suggest implementing it as an actual part of an in-depth system of the Site's Maintenance Routines(SMR) with more creative events it could cause to make the gameplay more busy for everyone.

REM would still be a thing but now in order directly influence the site it will be heavily reliant on the crews' site maintenance(Site Stability and Containment Zone Stability(one per SCP and Zone) with percentage based meters to keep track off that could be influenced in-game by antags/employee's with either way to harm or improve the site/containment zones stabilities which in turn would give engineering along with civilian and research jobs that are necessary to look after from time to time. Decreasing stability of different meters would cause much more drastic events that the REM could cause like dangerous work hazards or in case of the SCP Containment units when the stability degrades too low there's a high chance of Containment Failure Event occuring.

### Goals
- Provide depth to site maintenance.

- Benefits to site maintenance(research,positive events?).

- Give maintenance crew a clear task to accomplish–resolving stability issues and fixing hazards created.

- Give maintenance(Research,Engineering,In-zone Civilian) departments a general overview of the stability of the containments and zones.

- Give antags more subtle ways to fuck with security measures without directly interfacing with SCP's

### Non-goals

- We will try to not render containment breaches impossible if there's no antags/traitors or atleast leave it up to the player's decision by vote.

## Example of the feature
REM becomes dependant on percentage based meters of "zone stability" that are tied down to seperate places of the site with a corresponding computer and stability sensor that is capable of interfacing with the meter. It should become capable of causing more anomalous, but not hostile things to happen beyond facility whenever village is added.

Untill further notice these machines shouldn't be modifiable, destroyable, moveable and power going out shouldn't affect them because they're that important unless stated otherwise specifically.

1. Per SCP unique containment computer  within/close proximity to the containment zones specifically only capable of being interfaced with by either research personnel/containment specialist/chief engineer(or antagonist with an emag or stolen card) capable of both rising the stability, stabilizing the hazardous environments(to stop new events from happening without improving the stability outright) and degrading stability.
2. A variation of this containment computer designed to be deployed per general zone such as HCZ, EZ, LCZ to be interactable in office rooms by office workers and anyone above them. This thing would have also access to the wings' Stability meters and Interaction Records. Each Zone Commander should have a personal computer/laptop like this but incapable of working on the wing stability with them directly.
3. One stationary Stability/Interaction Overview computer in Administration, Research and Engineering. Site director could have a mobile version of this application.
4. A stability sensor that is in essence just a device that could be destroyed, turned off and repaired easily, this device would relay the zone stability information to the connected zone computers and to the general zone computer.

Civilians/Office workers could look after the generic zones like HCZ,EZ,LCZ these zones which in themselves when the stability lost they wouldn't _immediately_ impede the specific SCP containment zones aside from possibly CDZ that is present within LCZ but could prove grave when entire zones are without lights, door jams(not bolted), occasional slight electrocutions and increasingly hazardous environments(pipe leaks).

Research team could be delegated to look after the specific containment zones computers working on them to increase stability giving further more reason for the research team to stick around near the containments.

Engineering team could look out for the potential fall outs of these stability measures failing that will need fixing and could act as an intermediary to fill in if the research team seems lacking.

Further more I recognize that introducing stability systems could have a heavy influence on any other off-site hostile or positive events that could be automated to happen.

### Stability Meters and how REM could interact with them:
Preferably there should be no set in stone speed at which the stability is lost and it should be up to the popular player vote to choose at which speed the stability is decreased. Furthermore each Zone or SCP containment should have varying speeds at which they'll lose stability to guarantee a decent flow of the round.

**There should be a vote for multiple "Site-stresses" where Insane-Stress Event manager would still work like in its previous iterations where it could just randomly cause containment breaches to overwhelm crew and even every hand on deck helping out probably wouldn't save the site by-passing the stability mechanics for more chaos or that.**

**_Maybe for now 100% to 0% should take about 45 minutes for each SCP containment zone. I don't know. :^)_**

#### List of work hazards(some of which don't exist in the game yet???)

Filth Creation. 
Flickering lights. 
Lights now change to low emergency lighting but now slowly pulsing/blinking.
Lights sending out sparks. 
Lights shutting down for a second before turning back on stronger or weaker and changing in hue.
Lights emiting blue electric light then breaking.
Wet floor.
Door jams(ones that could be cleared with just a hand(Preferably gloved to not be hurt)).
Gas leak outbursts in varying severity(from CO2 to N2O or just hot steam) accompanied by tiles breaking above those pipes.

##### ≥95%
No chance of any event from happening

##### <95% ≥ 80%
Low chance: Flickering lights.

##### <80% ≥ 70% _Announcement about decreasing stability, malfunctioning power systems and work hazards occuring._
Low chance: Lights sending out sparks. Flickering lights.
Minimal chance: Wet floor. Filth Creation.

##### <70 ≥ 60 
Medium chance: Flickering lights. 
Low chance: Lights shutting down for a second before turning back on stronger or weaker and changing in hue. Lights sending out sparks.
Minimal chance: Lights breaking. Wet floor. Filth Creation.

##### <60 ≥ 40% _Announcement about increasingly dangerous work hazards._
Medium chance: Lights shutting down for a second before turning back on stronger or weaker and changing in hue. Wet floors.
Low chance: Door jams, Lights emiting very strong light then breaking. Filth Creation.
Minimal chance: Small gas leak outbursts(from CO2 to N2O or just hot steam) accompanied by tiles breaking above those pipes. 

##### <40 ≥ 30% _Announcement about the zone's structures weakening._
High chance: Lights shutting down for a second before turning back on stronger or weaker and changing in hue.
Medium chance: Dangerous gas leak outbursts(from CO2 to N2O or just hot steam) accompanied by tiles breaking above those pipes. Door jams.
Low chance: Slowly deteriorating Walls, Doors, Windows and Tiles. Filth Creation.
##### <30% ≥ 20% _Announcement about the zone's emergency state and civilian evacuation procedures._
Guaranteed: 
Lights now change to low-energy emergency lighting but now slowly pulsing/blinking.
In case of zones not containing SCP objects or directly interacting with D class, maintenance and regular(or if we get them evacuation shelters) doors would be bolted open.
Halt of any new hazards from occuring.
##### <20% ≥ 1% _Announcement about the zone's critical instability outbreak or in case of SCP containment, Imminent Breach._
Guaranteed: Doors return to normal operation.
High chance: Door jams.
Medium chance: Loss of 3% stability. Slowly deteriorating Walls, Doors, Windows and Tiles.
Low chance: Lights breaking. APC's stopping working.
##### <1%  _Announcement about Zone/Containment cell's shutdown._
Every electronic device is disabled aside from computers relating to zone stability
Every malfunctioning action is ceased but prolonged lack of stability should harm other surrounding/connected zones stability and all electric devices require a reset.

## Alternatives
- Instead of stability computers being only related to key access maybe it should only or additionaly rely on skills like bureaucracy and science in terms of efficiently using them. ex. General zones depend on basic beaurocracy meanwhile scp zones rely on beaurocracy basic and science trained or construction trained+ *shrug*.
- Add more physical (unique) containment components to scp zones that can malfunction, each being directly tied with random event manager, 173's shitting would be a drastic example but containment zones with complex devices outside of them influncing the breach would be a thing or like sending in food through a chute into the containment for like... 049 to eat.

## Potential Changes
- Probably refering to minute changes like "Why should Engineering and Research have access to computers capable of seeing stabilities instead of just administration having such things".
- Unique machine upgrade parts that don't rely on taking out parts from the existing machine
- The (how-it-could-work) meter is just a rough example and every containment cell would benefit from having a specific way it interacts with stability.

**Comment by TheTitanicEngineer** [comment]
Instead of relying on computers, we could make it like 173's breach system, in which we could adapt something to automatically and periodically provoke tiny random events that would require a certain response, failure to react to such events would increase the instability/maintenance status.

Basically, instead of having instability rely on computers, we could make it rely on the events themselves.

**Reply by ghost** [reply]
Indeed but if such events exist as similar to 173 shitting itself. Everyone would be capable of fixing it which more or less often fall once again unto the security to do all things by themselves, research will stay locked up in their ivory towers because actually the single best thing they can do at the moment is... doing research because testing SCP's is really not rewarding, engineering will atleast get something to do to atleast fake being busy,not irrevelant beyond round start-up. Office Workers will get "some" responsibility that could be easily handed over to wandering researchers/engineers.
049 will atleast SEE a researcher in their lifetime in the containment.

**Reply by ghost** [reply]
Although, going on the same vein, maybe everyone being able to do the 173 task is a good thing. I can't imagine many tasks relating to 106's containment to be anything but exclusive to engineering or research so a mix and mash with the tasks could be a good stepping stone.

**Reply by TheTitanicEngineer** [reply]
You got the idea

**Comment by aidan-bowman** [comment]
You could allow for antagonists to create workplace hazards themselves, that the random event manager would recognize and trigger if stability decreases. For example, a traitor might be able break into communications and perform some "sabotage" action on the servers, allowing for telecomms to suddenly go down if engineer starts losing stability.  Could result in some interesting strategies (sabotage everything you can, then wait for stability to go down. right when things start messing up, go loud. you'll have an advantage since you know what you sabotaged and what'll happen)
